src/Graph/GraphStd.cpp

```diff
@@ -248,6 +250,52 @@ void GraphStd<vid_t, eoff_t>::COOtoCSR() noexcept {
             _out_degrees[_coo_edges[i].first]++;
     }
 
+    if (_prop.is_directed_by_degree()) {
+        if (_prop.is_print())
+            std::cout << "Creating degree-directed graph ..." << std::endl;
+        eoff_t counter = 0;
+        vid_t u, v;
+        vid_t vid_small, vid_large;
+        degree_t deg_u, deg_v;
+        coo_t* coo_edges_tmp = new coo_t[_nE];
+        degree_t* _out_degrees_tmp = new degree_t[_nV]();
+        degree_t*  _in_degrees_tmp;
+        if (_structure.is_reverse())
+            _in_degrees_tmp = new degree_t[_nV]();
+        for (eoff_t i=0; i<_nE; i++) {
+            u = _coo_edges[i].first;
+            v = _coo_edges[i].second;
+            deg_u = _out_degrees[u];
+            deg_v = _out_degrees[v];
+            if ((deg_u > deg_v) || ((deg_u == deg_v) && (u > v))) {
+                coo_edges_tmp[counter++] = {u, v}; 
+            }
+        }
+        _coo_edges = coo_edges_tmp;
+        _nE = counter;
+        //std::cout << "Degree-directed graph has " << _nE << " edges" << std::endl;
```

Comment:
Should be removed. If this is a necessary debugging message then perhaps it can encapsulated in a #ifdef